lib/_http_agent.js

```diff
@@ -319,6 +330,20 @@ function installListeners(agent, s, options) {
   }
   s.on('close', onClose);
 
+  function onTimeout() {
+    debug('CLIENT socket onTimeout');
+
+    // Destroy if in free list.
+    // TODO(ronag): Always destroy, even if not in free list.
```

Comment:
Does this TODO plan to be completed? From the current usage, it may be a breaking change, which makes `timeout` change from `idletimeout` to `datatimeout`.